source/s3_meta_request.c

```diff
@@ -271,17 +280,12 @@ int aws_s3_meta_request_init_base(
             goto error;
         }
 
-        /* but keep original message around for headers, method, etc */
-        meta_request->initial_request_message = aws_http_message_acquire(options->message);
     } else if (options->send_async_stream != NULL) {
-        /* Read from async body-stream, but keep original message around for headers, method, etc */
         meta_request->request_body_async_stream = aws_async_input_stream_acquire(options->send_async_stream);
-        meta_request->initial_request_message = aws_http_message_acquire(options->message);
-        meta_request->maximize_async_stream_reads = options->maximize_async_stream_reads_internal_use_only;
 
-    } else {
-        /* Keep original message around, we'll read from its synchronous body-stream */
-        meta_request->initial_request_message = aws_http_message_acquire(options->message);
+    } else if (options->send_using_async_writes == true) {
+        meta_request->request_body_using_async_writes = true;
+        aws_byte_buf_init(&meta_request->async_write.data_buffer, allocator, 0);
```

Comment:
is 0 a good starting size for this?

Reply:
Yeah, 0 is a good starting size. We'll allocate when they finally get around to calling write(). In my experience, that's the way to go. It's totally worth debating the optimal *growth* strategy for a buffer once it has non-zero bytes, but that's for the followup PR